lib/iris/fileformats/pp_rules.py

```diff
@@ -162,12 +162,22 @@ def _convert_vertical_coords(lbcode, lbvc, blev, lblev, stash,
                             attributes={'positive': 'down'})
         coords_and_dims.append((coord, dim))
 
-    # Soil level.
-    if len(lbcode) != 5 and \
-            lbvc == 6:
-        coord = _dim_or_aux(model_level_number, long_name='soil_model_level_number',
-                            attributes={'positive': 'down'})
-        coords_and_dims.append((coord, dim))
+    # Deep soil level/depth.
+    if len(lbcode) != 5 and lbvc == 6:
+        if np.all(brsvd1 == brlev) and np.sum(brlev) == 0:
```

Comment:
Playing devil's advocate :smiling_imp: ... this would pass if brsvd1 = [-1, 1] and brlev = [-1, 1]. Perhaps better if the second condition was `not np.any(brlev)`?